myhoard/restore_coordinator.py

```diff
@@ -374,6 +374,7 @@ def apply_binlogs(self):
         initial_round = binlogs[0]["adjusted_remote_index"] == 1
         final_round = binlogs[-1]["adjusted_remote_index"] == self.pending_binlogs[-1]["adjusted_remote_index"]
         if initial_round and not mysql_started:
+            self._purge_old_slave_data()
```

Comment:
Is it relevant that this is done here? It would seem safer to do all the relay log manipulation while MySQL is stopped and do this just before we used to start MySQL at line 411

Reply:
you are right, will change that.